pallets/pallet-bonded-coins/src/traits.rs

```diff
@@ -0,0 +1,7 @@
+use frame_support::dispatch::DispatchResult;
+
+pub trait FreezeAccounts<AccountId, AssetId> {
+	fn freeze(caller: AccountId, who: AccountId, asset_id: AssetId) -> DispatchResult;
```

Comment:
None of them needs to be owned.
```suggestion
	fn freeze(caller: &AccountId, who: &AccountId, asset_id: &AssetId) -> DispatchResult;
```